Tasks/KubernetesManifestV0/src/actions/createSecret.ts

```diff
@@ -20,8 +20,17 @@ export async function createSecret() {
 }
 
 let getDockerRegistrySecretArgs = () => {
-    let dockerRegistryAuth = tl.getEndpointAuthorization(TaskInputParameters.dockerRegistryEndpoint, false).parameters;
-    return `docker-registry ${TaskInputParameters.secretName.trim()} --docker-username=${dockerRegistryAuth["username"]} --docker-password=${dockerRegistryAuth["password"]} --docker-server=${dockerRegistryAuth["registry"]} --docker-email=${dockerRegistryAuth["email"]}`
+    var registryType = tl.getInput("containerRegistryType", true);
+    if (registryType === "Azure Container Registry") {
+        let spnId = tl.getEndpointAuthorizationParameter(TaskInputParameters.dockerRegistryEndpoint, 'serviceprincipalid', true);
```

Comment:
Can we have 1 copy of this code across docker/kubernetes manifest task?

Reply:
@bansalaseem moved the logic to `docker-common`